src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -237,6 +237,34 @@ def cancelWorkflow(self, wf):
         elements = [x['id'] for x in data.get('rows', []) if x['key'][1] not in nonCancelableElements]
         return self.updateElements(*elements, Status='CancelRequested')
 
+    def updateSiteLists(self, wf, siteWhiteList=None, siteBlackList=None):
+        """
+        Update site list parameters in elements matching a given workflow and a list of element statuse
+
+        :param wf: workflow name
+        :param siteWhiteList: optional list of strings, new site white list
+        :param siteBlackList: optional list of strings, new site black list
+        :return: None
+        """
+        # Update elements in Available status
+        data = self.db.loadView('WorkQueue', 'jobStatusByRequest',
+                                {'reduce': False})
+        states = ['Available']
+        elementsToUpdate = [x['id'] for x in data.get('rows', []) if x['key'][-1] in states]
```

Comment:
You need to filter elements that match the workflow name as well, otherwise in practice you would be updating **everything** under Available status in a given agent.

This url can give you an example output:
https://cmsweb-preprod.cern.ch/couchdb/workqueue/_design/WorkQueue/_view/jobStatusByRequest?reduce=false

Reply:
ok